index.bs

```diff
@@ -552,13 +552,52 @@ spec: XHR; urlPrefix: https://xhr.spec.whatwg.org/
 
   Cross-origin messaging via <a>`postMessage`</a> requires that the recipient be
   able to see the suborigin namespace of the message sender so it can make an
-  appropriate access control decision. When a message is sent from a suborigin
-  namespace, the receiver has the `event.origin` value set to the serialized
-  suborigin value instead of the serializied origin, as described in
-  [[#serializing]]. Additionally, a new `suborigin` property must be added to
-  the `MessageEvent` given to the receiver which contains the suborigin
-  namespace value.
+  appropriate access control decision. We introduce a new dictionary
+  `EventInitiator` that holds information about both the origin and
+  the suborigin namespace:
+
+  <pre class="idl">
+    dictionary EventInitiator {
+      USVString origin = "";
+      USVString? suborigin = null;
+    };
+  </pre>
+
+  And we extend the definition of the MessageEvent and MessageEventInit as follows:
```

Comment:
Can you add a "We're monkey-patching HTML, submit a PR when we're sure this is the general direction we want to go in."  issue in this section?